File: TaskScheduler/TaskServer.py
__author__ = 'Brandon Maier'

'''
' Serves up tasks
'''

class TaskServer:
    class Task:
        time_left = 0
        def __init__(self, computation, deadline):



    waiting_tasks = None
    period = 0
    computation = 0
    deadline = 0

    def __init__(self, waiting_tasks, period, computation, deadline):
        self.waiting_tasks = waiting_tasks
        self.period = period
        self.computation = computation
        self.deadline = deadline

    def run(self):
        waiting_tasks.push()

File: Server/TaskServer.java
package TaskScheduler;

/**
 * Created by Brandon on 11/27/2014.
 */
public class TaskServer {
    final long period;
    final long runtime;
    final long relative_deadline;
    long virtual_time; // Time, where 0 occurs at server creation.

    public TaskServer(int period, int computation, int deadline) {
        virtual_time = 0;
        this.period = period;
        this.runtime = computation;
        this.relative_deadline = deadline;
    }

    public Task increment(long real_time) {
        Task task = null;

        if(virtual_time % period == 0)
            task = new Task(this, real_time + relative_deadline, runtime);

        virtual_time++;
        return task;
    }
}
